src/components/molecules/Visualizer/Engine/Cesium/Box/Edge.tsx

```diff
@@ -0,0 +1,129 @@
+import {
+  ArcType,
+  CallbackProperty,
+  Cartesian3,
+  Color,
+  ColorMaterialProperty,
+  Matrix4,
+  TranslationRotationScale,
+} from "cesium";
+import { FC, memo, useCallback, useEffect, useRef, useState } from "react";
+import { Entity, PolylineGraphics } from "resium";
+
+import { EventCallback } from "@reearth/util/event";
+
+import { useContext } from "../../../Plugin";
+
+export type EdgeEventCallback = EventCallback<
+  [
+    event: any,
+    edgeEvent: {
+      index: number;
+    },
+  ]
+>;
+
+export type EdgeProperties = { start: Cartesian3; end: Cartesian3; isDraggable?: boolean };
+
+type Props = {
+  id: string;
+  index: number;
+  edge: EdgeProperties;
+  trs: TranslationRotationScale;
+  isHovered: boolean;
+  fillColor?: Color;
+  hoverColor?: Color;
+  width?: number;
+  onMouseDown?: EdgeEventCallback;
+  onMouseMove?: EdgeEventCallback;
+  onMouseUp?: EdgeEventCallback;
+  onMouseEnter?: EdgeEventCallback;
+  onMouseLeave?: EdgeEventCallback;
+};
+
+export const Edge: FC<Props> = memo(function EdgePresenter({
+  id,
+  index,
+  edge,
+  isHovered,
+  fillColor,
+  trs,
+  width,
+  hoverColor,
+  onMouseDown,
+  onMouseMove,
+  onMouseUp,
+}) {
+  const ctx = useContext();
+  const [cbp] = useState(
+    () =>
+      new CallbackProperty(() => {
+        const position1 = new Cartesian3();
+        const position2 = new Cartesian3();
+        const matrix = Matrix4.fromTranslationRotationScale(trs);
+        Matrix4.multiplyByPoint(matrix, edge.start, position1);
+        Matrix4.multiplyByPoint(matrix, edge.end, position2);
+        return [position1, position2];
+      }, false),
+  );
```

Comment:
Sorry maybe a really stupid question/request, but just want to ask about this pattern (never seen useState used in this way before). Does this basically run the callback inside the useState() on render and assign it to the value (in this case, cbp)??
If so, why are you using useState like this, and not using useMemo?

Reply:
This indicates `cbp` is immutable. I want to create `CallbackProperty` only once. 
If we use `useMemo`, we need to set `deps array` to second args, and this will cause re-creating `CallbackProperty` and re-rendering. As a result, this will degrade performance.
But if you have any idea to improve this approach, I want to use it :eyes:

Reply:
Gotcha. Yeah the only other thing I can think of is just using useMemo with the `// eslint-disable-next-line react-hooks/exhaustive-deps` and then just have an empty deps array. But perhaps your approach is nicer since we don't have to disable anything.

But...do you actually not want `cbp` updated even if either `trs` or `edge` change?

Reply:
Yes, I don't want to update `cbp` even if `trs` or `edge` is changed.
Because if `cbp` is updated, the entity component of recium is re-rendered. `trs` is changed too many times, so updating `cbp` occurs too many re-rendering in the entity component, then the screen is jaggy.